src/assisted_installer_controller/assisted_installer_controller_test.go

```diff
@@ -1098,7 +1098,27 @@ var _ = Describe("installer HostRoleMaster role", func() {
 				assistedController.UpdateNodeLabels(context.TODO(), &wg)
 				wg.Wait()
 			})
+			It("Set label only in Ready node", func() {
+				nodeLabels := `{"node.ocs.openshift.io/storage":""}`
+				k8sNodes := GetKubeNodes(kubeNamesIds)
+				for i, cond := range k8sNodes.Items[0].Status.Conditions {
+					if cond.Type == v1.NodeReady {
+						k8sNodes.Items[0].Status.Conditions[i].Status = v1.ConditionFalse
+					}
+				}
+
+				hosts := create3Hosts(models.HostStatusInstalled, models.HostStageDone, nodeLabels)
+				mockbmclient.EXPECT().GetHosts(gomock.Any(), gomock.Any(), []string{models.HostStatusDisabled, models.HostStatusError}).
+					Return(hosts, nil).Times(2)
+				mockk8sclient.EXPECT().ListNodes().Return(k8sNodes, nil).Times(1)
+				mockk8sclient.EXPECT().ListNodes().Return(GetKubeNodes(kubeNamesIds), nil).Times(1)
+				mockk8sclient.EXPECT().PatchNodeLabels(gomock.Any(), nodeLabels).Return(nil).Times(2)
+				mockk8sclient.EXPECT().PatchNodeLabels(gomock.Any(), nodeLabels).Return(nil).Times(1)
```

Comment:
```suggestion
				mockk8sclient.EXPECT().PatchNodeLabels(gomock.Any(), nodeLabels).Return(nil).Times(3)
```
?

Reply:
The idea is to split logic and show that first time we will patch 2 nodes and on the next loop we will patch 1 node